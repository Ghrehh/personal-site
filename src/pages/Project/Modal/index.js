// @flow
import * as React from 'react';

import * as pathConstants from 'constants/paths';
import Page from 'components/Page';
import PageHeading from 'components/PageHeading';
import TransparentButton from 'components/TransparentButton';
import UnstyledLink from 'components/UnstyledLink';
import SlideUp from 'components/animation/SlideUp';

import styles from './styles.module.css';

type Props = {
  backgroundColor: string,
  heading: string,
  children: React.Node
};

const Modal = (props: Props) => (
  <SlideUp>
    <Page
      className={styles.page}
      style={{ backgroundColor: props.backgroundColor }}
      displayFooter={false}
    >
      <div className={styles.closeButtonContainer}>
        <TransparentButton>
          <UnstyledLink link={pathConstants.PROJECTS}>
            <span className={styles.closeButton}>Ã—</span>
          </UnstyledLink>
        </TransparentButton>
      </div>

      <PageHeading className={styles.title}>{props.heading}</PageHeading>

      {props.children}
    </Page>
  </SlideUp>
);

export default Modal;
