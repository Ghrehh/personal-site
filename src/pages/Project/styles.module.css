.page {
  padding-top: 0;
}

.closeButtonContainer {
  text-align: right;
  margin-bottom: 10px;
}

.closeButton {
  color: var(--charcoal);
  font-size: 50px;
  padding-right: 13px;
  transition: filter 0.5s;
}

.closeButton:hover {
  filter: brightness(1.5);
}

.body {
  text-align: center;
}

.description {
  margin-bottom: 40px;
  max-width: 500px;
  margin: auto;
  padding: 0 20px;
}

.links {
  margin-bottom: 40px;
}

.linkButtonContainer {
  text-align: center;
}
