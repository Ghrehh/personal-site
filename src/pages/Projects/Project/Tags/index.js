// @flow
import React from 'react';

import styles from './styles.module.css';

type Props = {
  tags: string[]
};

const Tags = (props: Props) => (
  <ul className={styles.tags}>
    {props.tags.map(tag => <li className={styles.tag}>{tag}</li>)}
  </ul>
);

export default Tags;
