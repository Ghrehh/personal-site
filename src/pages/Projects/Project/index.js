// @flow
import React from 'react';

import * as projectModel from 'models/data/project';
import * as pathConstants from 'constants/paths';
import SectionHeading from 'components/text/SectionHeading';
import UnstyledLink from 'components/UnstyledLink';

import styles from './styles.module.css';

type Props = {
  project: projectModel.Type,
  path: string
};

const Project = (props: Props) => (
  <div className={styles.projectOuter}>
    <UnstyledLink link={`${pathConstants.PROJECTS}/${props.path}`}>
      <div
        style={{ backgroundColor: props.project.backgroundColor }}
        className={styles.project}
      >
        <div className={styles.projectInner}>
          <SectionHeading className={styles.projectName}>
            {props.project.name}
          </SectionHeading>

          <p className={styles.projectBody}>{props.project.description}</p>
        </div>
      </div>
    </UnstyledLink>
  </div>
);

export default Project;
