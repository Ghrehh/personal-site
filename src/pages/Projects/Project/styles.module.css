.project {
  max-width: 500px;
  background-color: white;
  text-align: center;
  transition: transform 0.5s, filter 0.5s;
  padding: 40px 40px 20px;
  box-shadow: var(--box-shadow);
  border: none;
  margin: 40px auto;
}

@media (max-width: 620px) {
  .project {
    margin: 20px;
  }
}

.projectName {
  color: white;
}

.projectBody {
  color: white;
  margin-bottom: 28px;
}

.button {
  background-color: transparent;
  border: 2px solid white;
  border-radius: 4px;
  text-align: center;
  width: 145px;
  padding: 10px;
  transition: transform 0.5s, background-color 0.3s, color 0.3s;
  margin-bottom: 28px;
}

.button:hover {
  filter: none;
  background-color: white;
  color: var(--charcoal);
}

@media (max-width: 520px) {
}
