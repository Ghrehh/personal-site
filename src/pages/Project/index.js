// @flow
import React from 'react';

import * as projectsModel from 'models/data/projects';
import * as projectRouterParams from 'models/project-router-params';
import SectionHeading from 'components/text/SectionHeading';
import LinkButton from 'components/buttons/Link';
import ButtonContainer from 'components/ButtonContainer';
import Modal from './Modal';

import styles from './styles.module.css';

type Props = {
  projects: projectsModel.Type,
  match: projectRouterParams.Type
};

class Project extends React.Component<Props> {
  project = this.props.projects[this.props.match.params.project];

  renderWebsiteButton() {
    if (this.project.websiteLink === undefined) return null;

    return (
      <LinkButton external={true} link={this.project.websiteLink}>
        Website
      </LinkButton>
    );
  }

  render() {
    return (
      <Modal
        backgroundColor={this.project.backgroundColor}
        heading={this.project.name}
      >
        <section className={styles.description}>
          <SectionHeading>Description</SectionHeading>
          <p className={styles.body}>{this.project.description}</p>
        </section>

        <section className={styles.links}>
          <SectionHeading>Links</SectionHeading>

          <ButtonContainer className={styles.linkButtonContainer}>
            <LinkButton external={true} link={this.project.gitHubLink}>
              GitHub
            </LinkButton>
            {this.renderWebsiteButton()}
          </ButtonContainer>
        </section>
      </Modal>
    );
  }
}

export default Project;
