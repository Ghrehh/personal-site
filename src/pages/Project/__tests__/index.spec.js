import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Project from '../';

jest.mock('data', () => ({
  projects: {
    'cool-project': {
      name: 'foo',
      description: 'bar',
      backgroundColor: 'red'
    }
  }
}));

describe('Project', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};

    const component = (
      <Project match={{ params: { project: 'cool-project' } }} />
    );

    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(component);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
