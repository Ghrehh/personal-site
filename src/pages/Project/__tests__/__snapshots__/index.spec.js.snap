// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project renders the component correctly 1`] = `
<Modal
  backgroundColor="var(--green)"
  heading="NSTS METRONOME"
  onClose={[Function]}
>
  <section
    className="description"
  >
    <SectionHeading>
      Description
    </SectionHeading>
    <p
      className="body"
    >
      A sleek web metronome capable of playing in non-standard time signatures. Built using Vue JS
    </p>
  </section>
  <section
    className="links"
  >
    <SectionHeading>
      Links
    </SectionHeading>
    <ButtonContainer
      className="linkButtonContainer"
    >
      <LinkButton
        external={true}
        link="https://github.com/Ghrehh/nsts-metronome"
      >
        GitHub
      </LinkButton>
      <LinkButton
        external={true}
        link="https://nsts-metronome.github.io/"
      >
        Website
      </LinkButton>
    </ButtonContainer>
    <Tags
      className="tags"
      tags={
        Array [
          "Vue",
          "Music",
          "CSS Animation",
        ]
      }
    />
  </section>
</Modal>
`;
